lambdas/backend-api/src/templates/domain/template/template-repository.ts

```diff
@@ -0,0 +1,231 @@
+    '#templateStatus': 'templateStatus',
+    '#updatedAt': 'updatedAt',
+    '#templateType': 'templateType',
+  };
+
+  let expressionAttributeValues: Record<string, string> = {
+    ':name': template.name,
+    ':message': template.message,
+    ':templateStatus': template.templateStatus,
+    ':updateAt': new Date().toISOString(),
+    ':not_yet_submitted': TemplateStatus.NOT_YET_SUBMITTED,
+    ':templateType': template.templateType,
+  };
+
+  if (template.subject) {
+    updateExpression.push('#subject = :subject');
+    expressionAttributeNames = {
+      ...expressionAttributeNames,
+      '#subject': 'subject',
+    };
+    expressionAttributeValues = {
+      ...expressionAttributeValues,
+      ':subject': template.subject,
+    };
+  }
+
+  const input: UpdateCommandInput = {
+    TableName: process.env.TEMPLATES_TABLE_NAME,
+    Key: {
+      id: templateId,
+      owner,
+    },
+    UpdateExpression: `SET ${updateExpression.join(', ')}`,
+    ExpressionAttributeNames: expressionAttributeNames,
+    ExpressionAttributeValues: expressionAttributeValues,
+    ConditionExpression:
+      'attribute_exists(id) AND #templateStatus = :not_yet_submitted AND #templateType = :templateType',
+    ReturnValues: 'ALL_NEW',
+    ReturnValuesOnConditionCheckFailure: 'ALL_OLD',
+  };
+
+  try {
+    const response = await client.send(new UpdateCommand(input));
+
+    return success(response.Attributes as Template);
+  } catch (error) {
+    if (error instanceof ConditionalCheckFailedException) {
+      if (!error.Item) {
+        return failure(
+          ErrorCase.TEMPLATE_NOT_FOUND,
+          `Template not found`,
+          error
+        );
+      }
+
+      if (error.Item.templateStatus.S !== TemplateStatus.NOT_YET_SUBMITTED) {
+        return failure(
+          ErrorCase.TEMPLATE_ALREADY_SUBMITTED,
+          'Can not update template',
+          error,
+          {
+            templateStatus: `Expected ${TemplateStatus.NOT_YET_SUBMITTED} but got ${error.Item.templateStatus.S}`,
+          }
+        );
+      }
+
+      if (error.Item.templateType.S !== template.templateType) {
+        return failure(
+          ErrorCase.CANNOT_CHANGE_TEMPLATE_TYPE,
+          'Can not change template templateType',
+          error,
+          {
+            templateType: `Expected ${error.Item.templateType.S} but got ${template.templateType}`,
+          }
+        );
+      }
+    }
+
+    return failure(
+      ErrorCase.DATABASE_FAILURE,
+      'Failed to update template',
+      error
+    );
+  }
+};
+
+const list = async (owner: string): Promise<ApplicationResult<Template[]>> => {
+  try {
+    const response = await client.send(
+      new QueryCommand({
+        TableName: process.env.TEMPLATES_TABLE_NAME,
+        KeyConditionExpression: '#owner = :owner',
+        ExpressionAttributeNames: {
+          '#owner': 'owner',
+        },
+        ExpressionAttributeValues: {
+          ':owner': owner,
+        },
+        Limit: 50,
```

Comment:
This is temporary. It's just to bring the same functionality to the API as the amplify db calls. We should probably look todo pagination or something similar in the future.